service/posts/views.py

```diff
@@ -11,20 +11,12 @@ class PublicPostsList(generics.ListAPIView):
     pagination_class = PostsPagination
     serializer_class = PostSerializer
     authentication_classes = (NodeBasicAuthentication,)
-    permission_classes = (IsAuthenticated,)
+    # No permission class
 
     def get_queryset(self):
-        node = self.request.user
-
-        if not node.share_posts:
-            return Post.objects.none()
-
-        queryset = Post.objects.filter(visibility="PUBLIC")
+        remote_node = self.request.user
```

Comment:
Yeah, if an API doesn't have its authentication class set to this, it won't know what `Node` it's talking to, if any. Our internal APIs use `SessionAuthentication`, which ties into the default Django authentication system.